lib/services/role.go

```diff
@@ -2079,6 +2098,46 @@ func (set RoleSet) EnhancedRecordingSet() map[string]bool {
 	return m
 }
 
+// CanCreateHostUser determines whether a user is allowed to have a
+// temporary user provisioned for them on a node
+func (set RoleSet) CanCreateHostUser(s types.Server) bool {
+	for _, role := range set {
+		result, _, err := MatchLabels(role.GetNodeLabels(types.Allow), s.GetAllLabels())
```

Comment:
`s.GetAllLabels` should be hoisted out of the loop, it actually has to build a new map usually, and it's not meant to change for a given `s` anyway.